pkg/kubectl/cmd/util/helpers.go

```diff
@@ -83,6 +83,58 @@ func checkErr(err error, handleErr func(string)) {
 	handleErr(msg)
 }
 
+// CheckCustomErr is like CheckErr except a custom prefix error
+// string may be provied to help produce more specific error messages.
+// For example, for the update failed case this function could be called
+// with:
+//    cmdutil.CheckCustomErr("Update failed", err)
+// This function supresses the detailed output that is produced by CheckErr
+// and specifically the field is erased and the error message has the details
+// of the spec removed. Unfortunately, what starts off as a detail message is
+// a sperate field ends up being concatentated into one string which contains
+// the spec and the detail string. To avoid significant refactoring of the error
+// data structures we just extract the required detail string by looking for it
+// after "}': " which is horrible but expedient.
+func CheckCustomErr(customPrefix string, err error) {
+	checkCustomErr(customPrefix, err, fatal)
+}
+
+func checkCustomErr(customPrefix string, err error, handleErr func(string)) {
+	if err == nil {
+		return
+	}
+
+	if errors.IsInvalid(err) {
+		details := err.(*errors.StatusError).Status().Details
+		for i, _ := range details.Causes {
+			c := &details.Causes[i]
+			s := strings.Split(c.Message, "}': ")
```

Comment:
Indeed. But I am not sure what else I can do without changing the underlying error structures to avoid the conflation of the message I need with the spec.